doc/development/architecture.rst

```diff
@@ -0,0 +1,315 @@
+
+A solver is defined as a class with multiple steps, each step has to declare its dependencies.
+Additional models can be executed before and after each step to modify the behavior and add additional physics.
+
+This allows to easily extend existing solvers with new physics without modifying the core solver implementation and promotes code reuse.
+
+.. note::
+
+    The implementation is still work in progress but only details may change.
+
+Modular Solver Architecture
+---------------------------
+
+The modular solver architecture briefly touched on in the previous section is a key feature of FoamAdapter and allows for flexible composition of complex multi-physics simulations.
+An example of the modular solver architecture is illustrated in the following diagram:
+
+.. mermaid::
+
+   flowchart TD
+        
+        subgraph MAIN ["Main Solver Loop"]
+            STEP1["Momentum Predictor<br/>Solve velocity equation"]
+            STEP2["Additional Physics Modules"]
+            STEP3["Pressure Corrector<br/>Ensure mass conservation"]
+            STEP4["Field Updates<br/>Correct U, φ, turbulence"]
+        end
+        
+        STEP1 --> STEP2
+        STEP2 --> STEP3
+        STEP3 --> STEP4
+        
+        %% Physics Extensions (simplified)
+        subgraph AddPhysics ["Additional Physics Modules"]
+            direction TB
+            POROSITY["Porosity"]
+            ROTATION["Rotating Reference Frame"]
+            BUOYANCY["Boussinesq Approximation"]
+        end
+        POROSITY -.-> STEP1
+        ROTATION -.-> STEP1
+        BUOYANCY -.-> STEP2
+        BUOYANCY -.-> STEP3
+
+        style MAIN fill:#E3F2FD
+        style AddPhysics fill:#E3F2FD
+        style STEP1 fill:#2196F3,color:#fff
+        style STEP2 fill:#FF9800,color:#fff
+        style STEP3 fill:#9C27B0,color:#fff
+        style STEP4 fill:#607D8B,color:#fff
+
+
+In this architecture, the main solver loop consists of core steps such as momentum prediction and pressure correction.
+Additional physics modules (e.g., porosity, rotation, buoyancy) can be plugged into the workflow to modify the behavior of these core steps.
+
+This modular design enables users to easily add or remove physics effects without altering the fundamental solver structure, promoting code reuse and maintainability.
+The additional physics modules need to define their dependencies and the runtime system ensures that they are executed at the correct point in the solver loop.
+
+.. note::
+
+    implementation is still work in progress
+
+Field and Model Initialization
+------------------------------
+
+
+As the fields and solver needs to be initialized in the correct order before the solver run, FoamAdapter provides a structured initialization phase to ensure that the necessary fields and models are properly set up.
+
+The fields and models are stored lazily at first with the dependencies and a DAG is solved to determine the correct initialization order.
```

Comment:
I don't understand the meaning of this sentence. I would say they are not **stored** lazily but initialized?